bundle/edu.gemini.pot/src/main/java/edu/gemini/pot/sp/memImpl/MemObservation.java

```diff
@@ -109,7 +109,13 @@ public int getObservationNumber() {
     }
 
     public void setObservationNumber(int number) {
-        _obsNumber = number;
+        getProgramWriteLock();
+        try {
+            _obsNumber = number;
+            ((ProgramData) getProgram().getDocumentData()).ensureMaxEqualToOrGreaterThan(number);
+        } finally {
+            returnProgramWriteLock();
+        }
```

Comment:
Still kind of scary that this is public. The comment on `ISPObservation` says

> It is the caller's responsibility to ensure that observation numbers are not duplicated.

but it seems like it might make sense to check internally ... is that possible? Or does merge have to deal with temporarily inconsistent programs?